### PR TITLE
fix: captcha detection

### PR DESCRIPTION
**Pull request by doubleface**
Some user on the Cozy hosted users get a captcha we cannot handle.
Here is a fix to output the good error code and not LOGIN_FAILED in this case.
We won't put this connector in maintenance mode since it still works for auto hosted users

### CODE REVIEW

**Comment by nicofrand** [comment]
I had the issue with the previous website but could not reproduce it anymore with the new version, I just hoped there would be no more captchas :p.

Did you test it though? It looks like it misses the braces around the if (now that there are several lines)

File: src/index.js [hunk 1]
```diff
@@ -84,7 +84,11 @@ function login(loginToken, requiredFields) {
                 try {
                     body = JSON.parse(body);
                     if (!body || !body.authenticationSuccess)
-                        err = new Error("Authentication failed");
+                        if (body.loginForm.includes('window.renderCaptcha()')) {
+                            return reject(new Error('USER_ACTION_NEEDED.CAPTCHA'));
```
Comment (nicofrand) [inline src/index.js]: Why do you return the reject here? Setting `err` to `new Error('USER_ACTION_NEEDED.CAPTCHA')` should be enough?

Comment (nicofrand) [reply src/index.js]: Also, this is weird: `body` is supposed to be a JSOn.parse result here

Comment (doubleface) [reply src/index.js]: If I set err to `new Error('USER_ACTION_NEEDED.CAPTCHA')` the code will reject with `LOGIN_FAILED`.

About `body`, it is JSON.parsed but body.loginForm still contains text which is actually html with javascript.

Comment (nicofrand) [reply src/index.js]: Ok for both and sorry for the first one, indeed.

**Comment by nicofrand** [comment]
Thanks, I pushed a few modifications (renamed the commit, updated the version and a few nits) and merged!

**Pull request merged**